csrc/ops/utils.cpp

```diff
@@ -190,29 +190,34 @@ std::vector<IterDomain*> mapMatmulOpIterDomains(
   std::vector<IterDomain*> mapping(out_size, nullptr);
   auto inp_size = (int64_t)input_domain.size();
 
+  // Input A to matmul: {*, M, K}
+  // Input B to matmul: {*, K, N}
+  auto kpos = input_role == MatmulRole::INPUT_A ? inp_size - 1 : inp_size - 2;
```

Comment:
Changes in this file are to accomodate the new Reduction output domain by mapping it to the k position in each operand.